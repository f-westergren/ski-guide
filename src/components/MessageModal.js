import React, { useState } from 'react';
import { Button, Form, FormGroup, Label, Input, Modal, ModalHeader, ModalFooter, ModalBody } from 'reactstrap';
import SkiGuideApi from '../SkiGuideApi';

const MessageModal = ({ btnText, id }) => {
  const [message, setMessage] = useState('')
  const [isError, setIsError] = useState(false);
  const [modal, setModal] = useState(false);

  const toggle = () => {
    setModal(!modal);
  }

  const handleChange = e => {
    setMessage(e.target.value)
  }

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      await SkiGuideApi.newMessage({
        'to_user_id': id,
        'content': message
      })
      toggle();
    } catch (err) {
      setIsError(true)
    }
  }
  
  return(
    <>
    <Button className="text-uppercase" color="success" onClick={toggle}>{btnText}</Button>
    <Modal isOpen={modal} toggle={toggle}>
      <ModalHeader>
        Create Message
      </ModalHeader>
      <Form className="m-2" onSubmit={handleSubmit}>
      <FormGroup>
        <Input 
          type="textarea" 
          name="content" 
          onChange={handleChange}
          id="content" />
      </FormGroup>
      <ModalFooter>
        <Button>Submit</Button>
        <Button color="danger" onClick={toggle}>Cancel</Button>
      </ModalFooter>
      
      {isError && <span className="text-danger">Can't send message.</span>}
    </Form>

    </Modal>
    </>
  )
}

export default MessageModal;