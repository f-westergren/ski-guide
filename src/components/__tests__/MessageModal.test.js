import React from "react";
import { render } from "@testing-library/react";
import MessageModal from "../MessageModal";

it("renders without crashing", function() {
  render(<MessageModal />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<MessageModal />);
  expect(asFragment()).toMatchSnapshot();
});
