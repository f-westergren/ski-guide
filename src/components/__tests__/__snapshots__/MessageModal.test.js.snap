// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="text-uppercase btn btn-success"
    type="button"
  />
</DocumentFragment>
`;
